src/main/java/org/jetbrains/bsp/bazel/server/Server.java

```diff
@@ -47,25 +45,15 @@ public static void main(String[] args) {
       System.setOut(logStream);
       System.setErr(logStream);
 
-      BazelBspServerConfig config = BazelBspServerConfig.from(args);
-      BazelBspServer bspServer = new BazelBspServer(config);
+      BspIntegration bspIntegration = new BspIntegration(stdout, stdin, executor, traceWriter);
+      BazelBspServerConfig serverConfig = BazelBspServerConfig.from(args);
+      BazelBspServer bspServer = new BazelBspServer(serverConfig);
+      bspServer.startServer(bspIntegration);
 
-      Launcher<BuildClient> launcher =
-          new Launcher.Builder()
-              .traceMessages(traceWriter)
-              .setOutput(stdout)
-              .setInput(stdin)
-              .setLocalService(bspServer)
-              .setRemoteInterface(BuildClient.class)
-              .setExecutorService(executor)
-              .create();
-      BuildClientLogger buildClientLogger = new BuildClientLogger(launcher.getRemoteProxy());
-      bspServer.setBuildClientLogger(buildClientLogger);
-      BepServer bepServer = new BepServer(bspServer, launcher.getRemoteProxy(), buildClientLogger);
-      bspServer.setBepServer(bepServer);
-      io.grpc.Server server = ServerBuilder.forPort(0).addService(bepServer).build().start();
+      io.grpc.Server server = bspIntegration.getServer().start();
```

Comment:
What exactly do you mean? This logic has been split and partially moved to ``BazelBspServer::startServer/integrateBsp`` to simplify the dependencies (this is one of the trickiest places) – above all we need to construct BepServer and do ``.addService(bepServer)`` in ``BazelBspServer`` initialisation, otherwise we'd need to have 2-3 getters not used elsewhere or, like earlier, some cyclical dependencies.

Reply:
It's a syntactical question, the usage of `io.grpc.Server` instead of importing this at the beggining

Reply:
Oh right, it was like this before, I think the problem is that ``Server`` is by default ``org.jetbrains.bsp.bazel.server``, i.e. the exact class we're in :D Maybe we should change the name of ``Server.java`` to something else to have this one as ``Server`` and avoid confustion? Wdyt?

Reply:
Sounds great

Reply:
Do you have some ideas as to the name? I'm not sure, maybe something like _ServerInitializer_, _ServerLauncher_?

Reply:
Both sound good picks, but I'd go with Initializer